website/identifiers/clients/crossref.py

```diff
@@ -132,20 +133,49 @@ def build_posted_content(self, preprint, element, status):
 
         return posted_content
 
+    def _process_crossref_name(self, contributor):
+        # Adapted from logic used in `api/citations/utils.py`
+        # If the user has a family and given name, use those
+        if contributor.family_name and contributor.given_name:
+            given = contributor.given_name
+            middle = contributor.middle_names
+            family = contributor.family_name
+            suffix = contributor.suffix
+        else:
+            names = impute_names(contributor.fullname)
+            given = names.get('given')
+            middle = names.get('middle')
+            family = names.get('family')
+            suffix = names.get('suffix')
+
+        given_name = ' '.join([given, middle]).strip()
+        surname = ' '.join([family, suffix]).strip()
+
+        given_stripped = remove_control_characters(given_name)
+        # For crossref, given_name is not allowed to have numbers or question marks
+        given_processed = ''.join(
+            [char for char in given_stripped if (not char.isdigit() and char != '?')]
+        )
+        surname_processed = remove_control_characters(surname)
```

Comment:
Is surname allowed to have digits and `?`?

Reply:
YES! Isn't that odd?

for surname:
`pattern '[^\d\?]*[^\?\s]+[^\d]*' for type '#AnonType_surname'`

for given_name:
`pattern '[^\d\?]*' for type '#AnonType_given_name'`

Reply:
Odd indeed. 🆗